application-task-manager-ui/src/main/resources/TaskManager/Code/TaskReportMacro.xml

```diff
@@ -352,6 +241,15 @@
 #set ($displayColumns = $wikimacro.parameters.columns)
 #set ($rowCount = $wikimacro.parameters.limit)
 #set ($tableId = $escapetool.xml($wikimacro.parameters.id))
+#set ($spaceFilter = [])
+#foreach($space in $spaces.split('\s*,\s*'))
+  #set ($spaceRef = $services.model.resolveDocument($space))
+  #if ($spaceRef.name == 'WebHome')
+    #set ($discard = $spaceFilter.add($services.model.serialize($spaceRef.parent)))
+  #else
+    #set ($discard = $spaceFilter.add($services.model.serialize($spaceRef)))
+  #end
+#end
```

Comment:
```suggestion
#foreach($page in $pages.split('\s*,\s*'))
  #set ($pageRef = $services.model.resolveDocument($page))
  #set ($discard = $spaceFilter.add($services.model.serialize($pageRef.lastSpaceReference)))
#end
```

If picker for the ``pages`` macro parameter allows terminal pages to be selected, then with your approach the tasks from the terminal page won't be included. The suggested approach will include them, but also all the tasks from the sibling pages. So either limit the picker selection to nested pages or accept the limitation that selecting a terminal page will include the tasks from sibling pages. But in both cases I think you should simply use ``lastSpaceReference``.